### ISSUE TITLE
Trying to reproduce, some questions.

### ISSUE THREAD

**Issue by SamGalanakis** [MStypulkowski/CIF#3]
Hello, I'm working on reproducing your results in my own implementation as I wish to apply it to my own problem. I would appreciate some clarification on the following points.

1. In the paper you refer to using resnets but in the code you have several "type_emb", what was used for the paper and do any of the other settings give better resutls (for e.g. densenets).
2. In the train script you import F_flow_new, G_flow_new but I can't seem to find a definition anywhere, what are these and do they also provide an improvement?
3. In the paper each layer is followed by a permutation, is this implemented by simply switching the concatenation order as in :
`w = torch.cat([h2, h1], dim=1)`?
3. Are the results in the paper produced while training on only one class at a time (e.g. airplanes) or on multiple classes simultaneously?

Thanks in advance.

**Comment by MStypulkowski** [comment]
Hello Sam,

1. We found out that DenseNets give slightly better results and they are used in the final version. ResNets are not significantly worse.
2. Sorry about that, these are artifacts. For now, you can omit to import them and change if-else statements in lines 172-184 to always use `G_flow` and `F_flow`. We will fix it soon.
3. Exactly.
4. Yes, just one class at a time. We trained 3 separate models.

If you have any more problems please let me know.

**Comment by SamGalanakis** [comment]
Thanks for the quick reply, very clear. Regarding 4, was the model not expressive enough to perform for more than one class at a time?

**Comment by MStypulkowski** [comment]
We didn't train any model on more than one class at a time due to limited time and resources. My intuition is that it could work with more VRAM space for a single batch and a little bigger architecture.

**Comment by SamGalanakis** [comment]
Alright, thanks again for taking the time.
